src/sql/platform/accounts/common/accountViewModel.ts

```diff
@@ -48,23 +53,61 @@ export class AccountViewModel {
 		// 2) For each provider, get the accounts
 		// 3) Build parameters to add a provider and return it
 		try {
-			const metadata = await this._accountManagementService.getAccountProviderMetadata();
-			const accounts = await Promise.all(metadata.map(async providerMetadata => {
-				try {
-					const accounts = await this._accountManagementService.getAccountsForProvider(providerMetadata.id);
-					return <AccountProviderAddedEventParams>{
-						addedProvider: providerMetadata,
-						initialAccounts: accounts
-					};
-				} catch (err) {
-					this._logService.warn(`Error getting accounts for provider ${providerMetadata.id} : ${err}`);
-					return undefined;
-				}
-			}));
+			let accounts = await this.getAccountsForProviders();
+			if (this._authenticationService) {
+				const sessionAccounts = await this.getAccountForSessions();
+				accounts = accounts.concat(sessionAccounts);
+			}
+
 			return coalesce(accounts);
 		} catch (err) {
 			this._logService.warn(`Error getting account provider metadata : ${err}`);
 			return [];
 		}
 	}
+
+	private async getAccountsForProviders(): Promise<AccountProviderAddedEventParams[]> {
+		const metadata = await this._accountManagementService.getAccountProviderMetadata();
+		const accounts = await Promise.all(metadata.map(async (providerMetadata) => {
+			try {
+				const accounts = await this._accountManagementService.getAccountsForProvider(providerMetadata.id);
+
+				return <AccountProviderAddedEventParams>{
+					addedProvider: providerMetadata,
+					initialAccounts: accounts
+				};
+			} catch (err) {
+				this._logService.warn(`Error getting accounts for provider ${providerMetadata.id} : ${err}`);
+				return undefined;
+			}
+		}));
+		return accounts;
+	}
+
+	private async getAccountForSessions(): Promise<AccountProviderAddedEventParams[]> {
```

Comment:
Signing into Copilot doesn't show newly added account in account list, requires ADS to be refreshed.

![image](https://github.com/microsoft/azuredatastudio/assets/13396919/c4008bdb-47c2-4d1e-a2af-3de84d9e737e)

Reply:
The account list isn't refreshing. I chained logic used by remove account to remove the github account after signing out, but it didn't work. Most likely because the github account is being hydrated by the session information retrieved from the authentication service and not the account management service, so it's not in the map or list holding the rest of the accounts.